packages/browser/test/integrations/linkederrors.test.ts

```diff
@@ -52,67 +53,66 @@ describe('LinkedErrors', () => {
       expect(spy.calledOnce).equal(true);
     });
 
-    it('should recursively walk error to find linked exceptions and assign them to the event', () => {
-      const event = {
-        exception: {
-          values: [],
-        },
-        message: 'foo',
-      };
+    it('should recursively walk error to find linked exceptions and assign them to the event', async () => {
+      const three: ExtendedError = new SyntaxError('three');
 
-      const one: ExtendedError = new Error('one');
       const two: ExtendedError = new TypeError('two');
-      const three: ExtendedError = new SyntaxError('three');
+      two.cause = three;
 
-      const originalException = one;
+      const one: ExtendedError = new Error('one');
       one.cause = two;
-      two.cause = three;
 
+      const originalException = one;
+      const backend = new BrowserBackend({});
+      const event = await backend.eventFromException(originalException);
       const result = linkedErrors.handler(event, {
         originalException,
       });
 
       // It shouldn't include root exception, as it's already processed in the event by the main error handler
-      expect(result!.exception!.values!.length).equal(2);
-      expect(result!.exception!.values![0].type).equal('TypeError');
-      expect(result!.exception!.values![0].value).equal('two');
+      expect(result!.exception!.values!.length).equal(3);
+      expect(result!.exception!.values![0].type).equal('Error');
+      expect(result!.exception!.values![0].value).equal('one');
       expect(result!.exception!.values![0].stacktrace).to.have.property('frames');
-      expect(result!.exception!.values![1].type).equal('SyntaxError');
-      expect(result!.exception!.values![1].value).equal('three');
+      expect(result!.exception!.values![1].type).equal('TypeError');
+      expect(result!.exception!.values![1].value).equal('two');
       expect(result!.exception!.values![1].stacktrace).to.have.property('frames');
+      expect(result!.exception!.values![2].type).equal('SyntaxError');
+      expect(result!.exception!.values![2].value).equal('three');
+      expect(result!.exception!.values![2].stacktrace).to.have.property('frames');
     });
 
-    it('should allow to change walk key', () => {
+    it('should allow to change walk key', async () => {
       linkedErrors = new LinkedErrors({
         key: 'reason',
       });
-      const event = {
-        exception: {
-          values: [],
-        },
-        message: 'foo',
-      };
 
-      const one: ExtendedError = new Error('one');
-      const two: ExtendedError = new TypeError('two');
       const three: ExtendedError = new SyntaxError('three');
 
-      const originalException = one;
-      one.reason = two;
+      const two: ExtendedError = new TypeError('two');
       two.reason = three;
 
+      const one: ExtendedError = new Error('one');
+      one.reason = two;
```

Comment:
@kamilogorek any reason for reverting this in https://github.com/getsentry/sentry-javascript/commit/418ae83772f79c1966d133b06ad1cbd3e738fb53

Reply:
I didn't change any tests in #418ae83772f79c1966d133b06ad1cbd3e738fb53, just unified the code style with no changes in the logic itself. Also copied the same tests to node package - https://github.com/getsentry/sentry-javascript/commit/418ae83772f79c1966d133b06ad1cbd3e738fb53#diff-4c3c51aefa8e193ce4fd8de20e74b194

The only change was the Error orders in one of the tests, because it failed when I merged your second PR (although logic seems fine now) - https://github.com/getsentry/sentry-javascript/blob/efa7a33c5c5f84f5f9af1978df499ec802f6ccac/packages/browser/src/integrations/linkederrors.ts#L67
Top to bottom, eg. `[Three, Two, One]`.